### PR TITLE
chore(ICP-Rosetta): add functionality to increase staking amount

### PR DESCRIPTION
**Pull request by NikolasHai**
This PR proposes the following changes:

1. Add a function for the ICP Rosetta Client to increase the stake of an existing neuron
2. Remove the legacy system tests for staking through rosetta

### CODE REVIEW

**Review by cgundy — APPROVED** [review]

File: rs/tests/src/rosetta_tests/tests/neuron_staking.rs [hunk 1]
```diff
@@ -1,787 +0,0 @@
-use crate::rosetta_tests::{
```
Comment (fsodre) [inline rs/tests/src/rosetta_tests/tests/neuron_staking.rs:1]: Why are we deleting this entire test suite?

Comment (NikolasHai) [reply rs/tests/src/rosetta_tests/tests/neuron_staking.rs:1]: That is what the epic about [refactoring system tests](https://dfinity.atlassian.net/browse/FI-1446?atlOrigin=eyJpIjoiNjQzODI3YjhiM2E3NDg5OGEyM2NhZTU1YzU0MjI0N2EiLCJwIjoiaiJ9) is about. We want to migrate them to pocket ic tests as the once that are currently in place are outdated and should be removed.

**Review by mbjorkqvist — APPROVED** [review]
LGTM, thanks @NikolasHai !

File: rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs [hunk 2]
```diff
@@ -66,6 +70,117 @@ fn test_create_neuron() {
...
+                env.network_identifier.clone(),
+                &(*TEST_IDENTITY).clone(),
+                RosettaCreateNeuronArgs::builder(staked_amount.into())
+                    .with_from_subaccount(from_subaccount)
+                    .with_neuron_index(neuron_index)
+                    .build(),
+            )
+            .await
+            .unwrap();
+        env.pocket_ic.tick().await;
```
[the remark this answers is not in the document]
Comment (NikolasHai) [reply rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs]: I removed the ticks.

**Pull request merged**